source/simulator/assembly.cc

```diff
@@ -1729,6 +1776,7 @@ namespace aspect
                           this,
                           advection_field,
                           global_field_range,
+                          viscosity_per_cell,
```

Comment:
I think you ought to write this as `std_cxx11::cref(viscosity_per_cell)` to make sure no copy happens here.